agent/api/ecsclient/client.go

```diff
@@ -123,21 +123,31 @@ func (client *APIECSClient) RegisterContainerInstance(containerInstanceArn strin
 
 func (client *APIECSClient) registerContainerInstance(clusterRef string, containerInstanceArn string, attributes []string) (string, error) {
 	registerRequest := ecs.RegisterContainerInstanceInput{Cluster: &clusterRef}
+	var registrationAttributes []*ecs.Attribute
 	if containerInstanceArn != "" {
+		// We are re-connecting a previously registered instance, restored from snapshot.
 		registerRequest.ContainerInstanceArn = &containerInstanceArn
+	} else {
+		// This is a new instance, not previously registered.
+		// Custom attribute registration only happens on initial instance registration.
```

Comment:
If that's the case we need to be really explicit on this behavior; customers may expect that modifying the attributes in `/etc/ecs/ecs.config` would cause those changes to be reflected through the ECS API and we'll want to make sure we document that it won't happen.